vars/abapEnvironmentPipelineStageCloneRepositories.groovy

```diff
@@ -8,20 +8,26 @@ import static com.sap.piper.Prerequisites.checkScript
 @Field Set GENERAL_CONFIG_KEYS = []
 @Field STAGE_STEP_KEYS = [
     /** Pulls Software Components / Git repositories into the ABAP Environment instance */
-    'abapEnvironmentPullGitRepo'
+    'abapEnvironmentPullGitRepo',
+    /** Checks out a Branch in the pulled Software Component on the ABAP Environment instance */
+    'abapEnvironmentCheckoutBranch'
 ]
 @Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus(STAGE_STEP_KEYS)
 @Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
 /**
- * This stage clones Git repositories / software components to the ABAP Environment instance
+ * This stage clones Git repositories / software components to the ABAP Environment instance and checks out the master branch
  */
 void call(Map parameters = [:]) {
     def script = checkScript(this, parameters) ?: this
 
     def stageName = parameters.stageName?:env.STAGE_NAME
 
     piperStageWrapper (script: script, stageName: stageName, stashContent: [], stageLocking: false) {
-        abapEnvironmentPullGitRepo script: parameters.script
+        if (not(parameters.script.commonPipelineEnvironment.configuration.runStage?.get("Prepare System"))) {
```

Comment:
I would propose to switch the if and else branch and have a positive check condition. This would improve the readability a bit

Reply:
Adapt documentation: if "Prepare System" is no executed, we assume that a system is configured and all Software Components are cloned.

Reply:
Ok, changed it to having a positive check condition. 
Regarding the documentation: I changed the `indtroduction.md` file and the `configuration.md` file to clarify the usage of each steps during the `Clone Repositories` stage. Could you re-check?